src/activityLog/ActivityTreeItem.ts

```diff
@@ -0,0 +1,67 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { AzExtParentTreeItem, AzExtTreeItem, IActionContext, TreeItemIconPath } from "@microsoft/vscode-azext-utils";
+import { ThemeColor, ThemeIcon, TreeItemCollapsibleState } from "vscode";
+import { localize } from "../utils/localize";
+import { ActivityBase } from "./Activity";
+
+export class ActivityTreeItem extends AzExtParentTreeItem {
+
+    public constructor(parent: AzExtParentTreeItem, public readonly activity: ActivityBase) {
```

Comment:
Should take the interface instead of a specific class:
```suggestion
    public constructor(parent: AzExtParentTreeItem, public readonly activity: ActivityType) {
```

Reply:
There may need to be a wrapper class built around `ActivityType` (or `Activity` if renamed), for tracking the done/error state, based on the calls to `onSuccess()` and `onError()`. This tree item can be that wrapper class.